File: api-cmssignature/api-mssclient/vodafonemssprovider/csharp/src/profile/VodafoneProfileResponse.cs
using System;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.crypto.alg;
using tr.gov.tubitak.uekae.esya.api.crypto.util;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.profile;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.profile
{
    public class VodafoneProfileResponse: IProfileResponse
    {
        public static DigestAlgorithm SIGNING_CERT_DIGEST_ALG = DigestAlgorithm.SHA_256;

        protected ProfileInfo profileInfo;

        protected ECertificate cert;

        MSSParams aParams;

        public VodafoneProfileResponse(ECertificate cert, MSSParams aParams) {
            this.cert = cert;
            this.aParams = aParams;
            this.profileInfo = generateProfileInfo(cert, aParams);
        }
        
        private ProfileInfo generateProfileInfo(ECertificate cert, MSSParams aParams) {
            ProfileInfo profileInfo = new ProfileInfo();

            profileInfo.SetSerialNumber(cert.getSerialNumberHex());
            profileInfo.SetIssuerName(cert.getIssuer().getCommonNameAttribute());
            byte [] issuerBytes =  cert.getIssuer().getEncoded();
            string issuerBase64 = Convert.ToBase64String(issuerBytes, 0, issuerBytes.Length);
            profileInfo.SetCertIssuerDN(issuerBase64);
            profileInfo.setMssProfileURI(aParams.GetMsspProfileQueryUrl());
            profileInfo.SetCertHash(Convert.ToBase64String(DigestUtil.digest(DigestAlg.SHA256, cert.getBytes())));
            profileInfo.setDigestAlg(SIGNING_CERT_DIGEST_ALG.Url);

            return profileInfo;
        }

        public Status getStatus()
        {
            return Status.REQUEST_OK;
        }

        public string getMSSUri()
        {
            return aParams.GetMsspSignatureQueryUrl();
        }

        public ProfileInfo getProfileInfo()
        {
            return profileInfo;
        }

        public ECertificate getCertificate()
        {
            return cert;
        }
    }
}

File: api-cmssignature/api-mssclient/vodafonemssprovider/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/vodafone/profile/VodafoneProfileResponse.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.profile;

import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.ESYARuntimeException;
import tr.gov.tubitak.uekae.esya.api.common.util.Base64;
import tr.gov.tubitak.uekae.esya.api.crypto.exceptions.CryptoException;
import tr.gov.tubitak.uekae.esya.api.crypto.util.DigestUtil;
import tr.gov.tubitak.uekae.esya.api.infra.mobile.Status;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.profile.IProfileResponse;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.DigestAlgorithm;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.MSSParams;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.ProfileInfo;

public class VodafoneProfileResponse implements IProfileResponse {

    public static DigestAlgorithm SIGNING_CERT_DIGEST_ALG = DigestAlgorithm.SHA_256;

    protected ProfileInfo profileInfo;

    protected ECertificate cert;

    MSSParams params;

    public VodafoneProfileResponse(ECertificate cert, MSSParams params) {
        this.cert = cert;
        this.params = params;
        this.profileInfo = generateProfileInfo(cert, params);
    }

    private ProfileInfo generateProfileInfo(ECertificate cert, MSSParams params) {
        try {
            ProfileInfo profileInfo = new ProfileInfo();

            profileInfo.setSerialNumber(cert.getSerialNumberHex());
            profileInfo.setIssuerName(cert.getIssuer().stringValue());
            profileInfo.setCertIssuerDN(Base64.encode(cert.getIssuer().getEncoded()));
            profileInfo.setMssProfileURI(params.getMsspProfileQueryUrl());
            profileInfo.setCertHash(Base64.encode(DigestUtil.digest(SIGNING_CERT_DIGEST_ALG.getAlgorithm(), cert.getEncoded())));
            profileInfo.setDigestAlg(SIGNING_CERT_DIGEST_ALG.getUrl());

            return profileInfo;
        } catch (CryptoException e) {
            throw new ESYARuntimeException("Can get generate profile info from certificate.", e);
        }
    }

    public String getMSSUri() {
        return params.getMsspSignatureQueryUrl();
    }

    public ProfileInfo getProfileInfo() {
        return profileInfo;
    }

    public ECertificate getCertificate() {
        return cert;
    }

    public Status getStatus() {
        return Status.REQUEST_OK;
    }
}
